### PR TITLE
add self claim decline docs

### PR DESCRIPTION
**Pull request by gmanningcr**
connects CorporateRewards/MyRewards#7653

### CODE REVIEW

File: source/includes/api/performance/_self_claims.md [hunk 1]
```diff
@@ -174,3 +174,84 @@ declined_reason | `string` | optional - a string indicating the reason for decli
...
+#### Request Parameters
+
+##### Body Parameters
+
+If decline reasons are active on the promotion, then one of `reason_for_decline_id` or `reason_for_decline_text`, but not both, should be provided when declining a claim.
+
+Parameter | Type | Description
+--------- | ---- | ----
+reason_for_decline_id | `integer` | Required conditional - The ID of a reason for decline belonging to the promotion.
+reason_for_decline_text | `string` | Required conditional - A text string of a reason for decline. Can only be provided if the promotion allows free text decline reasons.
```
Comment (njall) [inline source/includes/api/performance/_self_claims.md:257]: @gmanningcr The phrasing `Required conditional` could be confusing, maybe?

Comment (gmanningcr) [reply source/includes/api/performance/_self_claims.md:257]: @njall - Yeah, I struggled around how to word this. A claim requires a decline reason if decline reasons are active (either specific decline reasons set up or free text allowed as a decline reason). Then if the promo only has specific decline reasons, and not free text, then reason_for_decline_id is required. If the promo has specific reasons AND free text allowed, then one of reason_for_decline_id or reason_for_decline_text is required, but not both, and if only free text is allowed then only reason_for_decline_text is required. Any suggestions?

**Review by njall — APPROVED** [review]

File: source/includes/api/performance/_self_claims.md (hunk 1 above)
Comment (njall) [reply source/includes/api/performance/_self_claims.md:257]: I guess I'd consider exploding it out. So have different params tables for each possible state. But I think its okay with as it is

Comment (rjpaskin) [reply source/includes/api/performance/_self_claims.md:257]: "Required if enabled"?

**Review by rjpaskin — APPROVED** [review]

**Pull request merged**